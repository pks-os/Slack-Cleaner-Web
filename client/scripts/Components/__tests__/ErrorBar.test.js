import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ErrorBar from '../ErrorBar';

const DEFAULT_PROPS = {
  message: 'Something bad happened!',
  styles: {
    color: 'blue'
  }
};

describe('<ErrorBar />', () => {
  it('Renders Correctly', () => {
    const tree = renderer.create(<ErrorBar {...DEFAULT_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct message if the "present" prop is passed', () => {
    const tree = shallow(<ErrorBar {...DEFAULT_PROPS} present />);
    expect(tree.text().includes(DEFAULT_PROPS.message)).toBe(true);
  });
});