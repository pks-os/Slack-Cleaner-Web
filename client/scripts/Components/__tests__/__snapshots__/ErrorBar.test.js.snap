// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBar /> Renders Correctly 1`] = `
<div
  className="Error"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <p>
    ok bye!
  </p>
</div>
`;
